mpy-cross/main.c

```diff
@@ -309,6 +328,8 @@ MP_NOINLINE int main_(int argc, char **argv) {
                 } else {
                     return usage(argv);
                 }
+            } else if (strcmp(argv[a], "--") == 0) {
```

Comment:
Am I reading this right and `--` is required in order to use `-` as the input file?

IMHO, it would be nicer to leave out `--` since we don't really need it for anything else and add a case instead for matching `-` by itself.

Reply:
`--` is a common convention, and is required for any filenames starting with `-`. Not a particularly common use case, I admit, but essential for script hardening.

Another common convention is defaulting to stdin if no input file is given, maybe that would be a good alternative to having to specify `-- -`?

Reply:
> Another common convention is defaulting to stdin if no input file is given

That sounds fine to me.

Reply:
I think it would be good to also add a case to handle `-` explicitly, even when `option_parsing_active` is true.  (Eg `diff` allows passing `-` directly.)

Reply:
Done.